ibis/backends/pandas/tests/execution/test_timecontext.py

```diff
@@ -289,13 +289,73 @@ def adjust_context_custom_asof_join(
         right=time_keyed_right,
         predicates='time',
         by='key',
-        tolerance=4 * ibis.interval(days=1),
+        tolerance=ibis.interval(days=4),
     ).to_expr()
     expr = expr[time_keyed_left, time_keyed_right.other_value]
     context = (pd.Timestamp('20170105'), pd.Timestamp('20170111'))
     expr.execute(timecontext=context)
 
 
+def test_adjust_context_complete_shift(
```

Comment:
I guess at some point we should probably try to deduplicate this testing code where possible.

Reply:
agreed! For some tests it will be a little tricky due to small differences like using `dd` instead of `pd`, and calling `.compute()` in the Dask tests. It would be nice to go through and minimize this duplication eventually though

Reply:
+1 to deduplicated test. We should test Spark as well.

Reply:
@timothydijamco I don't understand the argument - why can this not be a shared tests?

Reply:
oh actually you're right, this can be shared. The usage of Dask-specific things is only to compute the expected result, but for a shared test we can compute the expected result using Pandas for all backends (which is what we normally do in our shared tests I believe)

Let me refactor

Reply:
Unfortunately none of our source datasets that can be used in our shared tests (i.e. the data defined in https://github.com/ibis-project/testing-data) have a `time` column, which is required when testing context adjustment code (the source table itself must have a `time` column—it cannot be added in a test to bypass this)

Since there is no shared dataset with a `time` column, each backend (Pandas, Dask, and PySpark) is currently defining test datasets using their respective backend DataFrame APIs:
- [Pandas](https://github.com/ibis-project/ibis/blob/master/ibis/backends/pandas/tests/execution/conftest.py#L131-L187)
- [Dask](https://github.com/ibis-project/ibis/blob/master/ibis/backends/dask/tests/execution/conftest.py#L126-L187)
- [PySpark](https://github.com/ibis-project/ibis/blob/master/ibis/backends/pyspark/tests/conftest.py#L235-L249)

If we add test data with a `time` column to https://github.com/ibis-project/testing-data, then we would be able to consolidate many of the separate tests that across Pandas/PySpark/Dask that use `time` column into shared tests

Would everyone be OK with this being a separate issue / PR to narrow the scope of this PR?

Reply:
ok with doing this as a separate issue / followup.

Reply:
Yep I am OK with that

Reply:
ok, I created a follow-up: https://github.com/ibis-project/ibis/issues/3109